foxpuppet/windows/browser/bookmarks/bookmark.py

```diff
@@ -0,0 +1,213 @@
+
+                    frame_element = self.selenium.find_element(
+                        *BookmarkLocators.ADD_BOOKMARK_FRAME
+                    )
+                    self.selenium.switch_to.frame(frame_element)
+                    if bookmark_data is not None:
+                        if bookmark_data["name"]:
+                            self.actions.send_keys(bookmark_data["name"]).perform()
+                        self.actions.send_keys(Keys.TAB).perform()
+
+                        if bookmark_data["url"]:
+                            self.actions.send_keys(
+                                bookmark_data["url"] + Keys.TAB
+                            ).perform()
+
+                        if (tags := bookmark_data["tags"]) is not None:
+                            for tag in tags:
+                                self.actions.send_keys(tag).perform()
+                                self.actions.send_keys(",").perform()
+                            self.actions.send_keys(Keys.TAB).perform()
+
+                        if bookmark_data.get("keyword"):
+                            keyword = bookmark_data["keyword"] or ""
+                            self.actions.send_keys(keyword + Keys.TAB).perform()
+
+                    self.actions.send_keys(
+                        Keys.TAB, Keys.TAB, Keys.TAB, Keys.ENTER
+                    ).perform()
+                    if (
+                        folder := self.selenium.find_element(
+                            *BookmarkLocators.BOOKMARK_FOLDER
+                        )
+                    ) is not None:
+                        folder.click()
+                        self.selenium.switch_to.frame(folder)
+                        self.actions.send_keys(Keys.TAB, Keys.ENTER).perform()
+
+    def retrieve_bookmark(self, label: str) -> bool:
+        """
+        Check if a bookmark with the given label exists.
+
+        Args:
+            label (str): The name of the bookmark to search for.
+        """
+        with self.selenium.context(self.selenium.CONTEXT_CHROME):
+            self.selenium.find_element(*BookmarkLocators.PANEL_MENU).click()
+            self.selenium.find_element(*BookmarkLocators.PANEL_BOOKMARK_MENU).click()
+            panel_bookmarks = self.selenium.find_element(
+                *BookmarkLocators.PANEL_BOOKMARK_TOOLBAR
+            )
+            menu_items = panel_bookmarks.find_elements(
+                By.CSS_SELECTOR, "toolbarbutton.bookmark-item"
+            )
+            if any(
+                label.lower() in item_label.lower()
+                for item in menu_items
+                if (item_label := item.get_attribute("label"))
+            ):
+                return True
+        return False
+
+    def delete_bookmark(
+        self, label: Optional[str] = None, is_detailed: bool = False
+    ) -> None:
+        """
+        Delete a bookmark using either quick delete (star button) or detailed menu approach.
```

Comment:
I wonder if this is overly complex? Does it matter how a bookmark is deleted?

Reply:
The only reason I added the more detailed approach is because if we test adding a bookmark using the more detailed approach, then we should do same with the delete.